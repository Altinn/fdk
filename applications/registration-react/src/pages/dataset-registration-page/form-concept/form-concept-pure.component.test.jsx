import React from 'react';
import { shallow } from 'enzyme';
import { FormConceptPure } from './form-concept-pure.component';

let defaultProps;
let wrapper;

beforeEach(() => {
  defaultProps = {
    syncErrors: {
      keyword: null
    }
  };
  wrapper = shallow(<FormConceptPure {...defaultProps} />);
});

test('should render FormConceptPure correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should render FormConceptPure correctly with syncErrors', () => {
  wrapper.setProps({
    syncErrors: {
      keyword: {
        nb: 'Feil'
      }
    }
  });
  expect(wrapper).toMatchSnapshot();
});
