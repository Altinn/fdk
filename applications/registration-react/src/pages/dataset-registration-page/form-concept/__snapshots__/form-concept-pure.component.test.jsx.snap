// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FormConceptPure correctly 1`] = `
<form>
  <div
    className="form-group"
  >
    <withStateHandlers(Helptext)
      term="Dataset_content"
      title="Begrep"
    />
    <Field
      component={[Function]}
      fieldLabel="no"
      label="Begrep"
      name="concepts"
      type="text"
    />
  </div>
  <div
    className="form-group"
  >
    <withStateHandlers(Helptext)
      term="Dataset_keyword"
      title="Søkeord"
    />
    <Field
      component={[Function]}
      fieldLabel="nb"
      label="Søkeord"
      name="keyword"
      type="text"
    />
  </div>
</form>
`;

exports[`should render FormConceptPure correctly with syncErrors 1`] = `
<form>
  <div
    className="form-group"
  >
    <withStateHandlers(Helptext)
      term="Dataset_content"
      title="Begrep"
    />
    <Field
      component={[Function]}
      fieldLabel="no"
      label="Begrep"
      name="concepts"
      type="text"
    />
  </div>
  <div
    className="form-group"
  >
    <withStateHandlers(Helptext)
      term="Dataset_keyword"
      title="Søkeord"
    />
    <Field
      component={[Function]}
      fieldLabel="nb"
      label="Søkeord"
      name="keyword"
      type="text"
    />
  </div>
</form>
`;
